### DISCUSSION TITLE
Duplicate ID while using a Parallels VM (Mac OS M-processor)

### DISCUSSION THREAD

**Discussion by Ndoll1** [rustdesk/rustdesk#6053] [General]
### Bug Description

**Describe the bug**
As a PoC we created some Mac virtual machines using Parallels. We wanted to hook into those using Rustdesk. While setting up Rustdesk generated identical ID's for different VM's. 

_The serial number on both VM's are the same, that is a thing that can happen on a VM, because in some instances you need to freeze an older version and upgrade a copy. Or change some parameters because you have the VM set-up differently for a customer, but the base is equal. Duplicating the VM is more convenient then._ 

**Describe the environment**
Rustdesk 1.2.2 - Mac OS ARM version
 - Running on a Parallels Mac OS VM

### How to Reproduce

Install an instance of Rustdesk Mac OS ARM on a parallels VM
In some instances a duplicate ID will be generated, to me this happened to imported VM's.
In those cases the common thing is that the serialnumber of the VM's were identical. See image above.

### Expected Behavior

It should have generated a different ID. What also would solve the issue is the option to reset/regenerate the ID of the Rustdesk instance.

### Operating system(s) on local side and remote side

Mac OS -> Not Applicable

### RustDesk Version(s) on local side and remote side

1.1.2  -> Not Applicable

### Screenshots

![image](https://github.com/rustdesk/rustdesk-server/assets/75546808/9612c85e-5a9e-4958-878f-c7e71c6ba317)
_For privacy reasons blurred the middle part of the ID._

### Additional Context

I tried changing the serial number of the VM's themselves, this is sadly not possible on M-processor macs as it stands right now
 Source: https://kb.parallels.com/123455

**Comment by TBG-FR** [comment]
@rustdesk This is a real issue, ID should never be duplicated, the server should refuse a duplicated client or send back a new unused id, no ?
